### PR TITLE
Improvements & Feature ..

### PR DESCRIPTION
**Pull request by cloned2k16**
proposing a re factory which includes some improvements 
and pointing out a BUG which may be a feature but the have to be handled properly ...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,36 +1,75 @@
...
+  , '000000000000'
+  , '0000000000000'
+  , '00000000000000'
+  , '000000000000000'
+  , '0000000000000000'  //16
+];
 
-  // doesn't need to pad
+function leftPad (str, len, ch) {
+  var   pad     = ''
```
Comment (alkuhar) [inline index.js:46]: @cloned2k16 did you mean to write the var definition like this?
`var pad = '', cache;`

Comment (cloned2k16) [inline index.js:46]: I prefer to write it as I did .. which is the same ... isn't ?

**Comment by stevemao** [comment]
The commit is huge. For people who are lazy to read the whole changes, The PR adds use-cases for

```
+  assert.strictEqual(leftPad('feature'  ,  8, 'bug or '), 'bug or feature' ); 
+  assert.strictEqual(leftPad('feature'  , 10, 'buggy ') , 'buggy buggy buggy feature' );
```

I don't know if anyone is using it.

**Comment by cloned2k16** [comment]
well , actually these two use case were supposed to highlight a BUG ..
which I didn't fix either ... because maybe can be converted in feature ..

the fact is ..  as it is right now, is just a plain and simple BUG

because anyone going to what a function that pads a string with another,
 expects it to count in a different (straightforward) way ..

eg ...

leftPad('feature'  ,  8, 'bug or ')  ..

should be ..
leftPad('feature'  ,  1, 'bug or ') 
instead ...

finally, If you just have a look to the whole file,
instead to just look to diff changes,
isn't so huge ...
while still true...
it changes a lot from original ...

[whole file](https://github.com/cloned2k16/left-pad/edit/master/index.js?pr=/stevemao/left-pad/pull/28)

**Comment by cloned2k16** [comment]
If you just look at the whole file instead of to diff changes isn't so huge ...
 while yes it changes a lot from original ...

**Comment by cloned2k16** [comment]
[whole file] (https://github.com/cloned2k16/left-pad/edit/master/index.js?pr=/stevemao/left-pad/pull/28)

File: index.js (hunk 1 above)
Comment (alkuhar) [inline index.js:46]: Doesn't the interpreter add a semicolon at the end of `cache`? So, you would end up with 2 semicolons:

```
var   pad     = ''
,     cache;
;
```

Comment (cloned2k16) [inline index.js:46]: sort of ... 
 but I prefer to add the last one for clarity of the code and better portability ..

it doesn't harm having useless semicolons 
(actually I dislike javascript ''''feature''' of ending line automatically )

**Pull request closed without merging**